From fbde1593948ae95f287b2167d0bec5b27948cc01 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
Date: Tue, 6 Aug 2024 17:04:55 +0200
Subject: [PATCH] Export all symbols of mjpg_streamer binary

Fixes runtime error with stripped binary

    dlopen: /usr/lib/mjpg-streamer/input_uvc.so: undefined symbol: resolutions_help

Source: http://lists.busybox.net/pipermail/buildroot/2024-August/759732.html

Upstream: https://github.com/jacksonliam/mjpg-streamer/pull/401

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 mjpg-streamer-experimental/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mjpg-streamer-experimental/CMakeLists.txt b/mjpg-streamer-experimental/CMakeLists.txt
index cf26620..3ff12cd 100644
--- a/mjpg-streamer-experimental/CMakeLists.txt
+++ b/mjpg-streamer-experimental/CMakeLists.txt
@@ -92,6 +92,7 @@ set (CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 add_executable(mjpg_streamer mjpg_streamer.c
                              utils.c)
 
+set_property(TARGET mjpg_streamer PROPERTY ENABLE_EXPORTS ON)
 target_link_libraries(mjpg_streamer pthread dl)
 install(TARGETS mjpg_streamer DESTINATION bin)
 
-- 
2.39.2

